ws/api/transport.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-zoo/bone"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gorilla/websocket"
+	"github.com/mainflux/mainflux"
+	manager "github.com/mainflux/mainflux/manager/client"
+	"github.com/mainflux/mainflux/ws"
+	broker "github.com/nats-io/go-nats"
+)
+
+const topic string = "src.*"
+
+var (
+	errUnauthorizedAccess = errors.New("missing or invalid credentials provided")
+	upgrader              = websocket.Upgrader{
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+		CheckOrigin: func(r *http.Request) bool {
+			return true
+		},
+	}
+	auth   manager.ManagerClient
+	nc     *broker.Conn
+	logger log.Logger
+)
+
+// MakeHandler returns http handler with handshake endpoint.
+func MakeHandler(svc ws.Service, mc manager.ManagerClient, bc *broker.Conn, l log.Logger) http.Handler {
+	auth = mc
+	nc = bc
+	logger = l
+
+	mux := bone.New()
+	mux.GetFunc("/channels/:id/messages", handshake(svc))
+
+	return mux
+}
+
+func handshake(svc ws.Service) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sub, err := authorize(r)
+		if err != nil {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		// Create new ws connection.
+		conn, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			logger.Log("error", fmt.Sprintf("Failed to upgrade connection to websocket: %s", err))
+			return
+		}
+		socket := ws.NewSocket(conn)
+
+		// Subscribe to all NATS subjects.
+		topic := fmt.Sprintf("%s.%s", topic, sub.ChanID)
+		brokerSub, err := nc.Subscribe(topic, func(msg *broker.Msg) {
+			if msg == nil {
```

Comment:
This does not belong here. I would expect callback to be the part of the service.